cmd/operator/main.go

```diff
@@ -265,6 +266,13 @@ func Main() int {
 		return 1
 	}
 
+	pao, err := prometheusagentcontroller.New(ctx, cfg, log.With(logger, "component", "prometheusagentoperator"), r)
```

Comment:
we might want to add `--prometheus-instance-selector` and `--prometheus-instance-namespaces` flags but it doesn't have to be in this PR.

Reply:
though it might be ok to use the same flags for both Prometheus and PrometheusAgent...